src/main/java/org/osiam/storage/query/GroupFilterExpression.java

```diff
@@ -0,0 +1,63 @@
+package org.osiam.storage.query;
+
+import org.osiam.resources.scim.Group;
+import org.osiam.storage.entities.GroupEntity;
+
+import java.util.Locale;
+
+public class GroupFilterExpression extends FilterExpression<GroupEntity> {
+
+    private final Field field;
+
+    public GroupFilterExpression(String field, FilterConstraint constraint, String value) {
+        super(constraint, value);
+        this.field = new Field(field);
+    }
+
+    @Override
+    public Field getField() {
+        return field;
+    }
+
+    public static class Field implements FilterExpression.Field<GroupEntity> {
+        private String fieldName;
+        private GroupQueryField queryField;
+
+        public Field(String field) {
+
+            if (field.startsWith(Group.SCHEMA)) {
```

Comment:
This still bothers me. I think we should be case-insensitive here. We are at least case-insensitive when checking the URN of an extension. I think this is because of the text in the [SCIM RFC, p. 17](http://tools.ietf.org/html/rfc7644#page-17):

>    Attribute names and attribute operators used in filters are case
>    insensitive.  For example, the following two expressions will
>    evaluate to the same logical value:
> 
>    filter=userName Eq "john"
> 
>    filter=Username eq "john"

But I don't know, whether this applies just to the attribute name or to the URN too. What do you think?